tests/unit/test_locales.py

```diff
@@ -13,98 +15,104 @@ class TestLocales(SearxTestCase):
+    @parameterized.expand(
+        [
+            'de',
+            'fr',
+            'zh',
+        ]
+    )
+    def test_locale_languages(self, locale: str):
         # Test SearXNG search languages
-
-        self.assertEqual(locales.match_locale('de', locale_tag_list), 'de')
-        self.assertEqual(locales.match_locale('fr', locale_tag_list), 'fr')
-        self.assertEqual(locales.match_locale('zh', locale_tag_list), 'zh')
-
+        self.assertEqual(locales.match_locale(locale, self.locale_tag_list), locale)
+
+    @parameterized.expand(
+        [
+            ('ca-es', 'ca-ES'),
+            ('de-at', 'de-AT'),
+            ('de-de', 'de-DE'),
+            ('en-UK', 'en-GB'),
+            ('fr-be', 'fr-BE'),
+            ('fr-be', 'fr-BE'),
+            ('fr-ca', 'fr-CA'),
+            ('fr-ch', 'fr-CH'),
+            ('zh-cn', 'zh-CN'),
+            ('zh-tw', 'zh-TW'),
+            ('zh-hk', 'zh-HK'),
+        ]
+    )
+    def test_match_region(self, locale: str, expected_locale: str):
         # Test SearXNG search regions
-
-        self.assertEqual(locales.match_locale('ca-es', locale_tag_list), 'ca-ES')
-        self.assertEqual(locales.match_locale('de-at', locale_tag_list), 'de-AT')
-        self.assertEqual(locales.match_locale('de-de', locale_tag_list), 'de-DE')
-        self.assertEqual(locales.match_locale('en-UK', locale_tag_list), 'en-GB')
-        self.assertEqual(locales.match_locale('fr-be', locale_tag_list), 'fr-BE')
-        self.assertEqual(locales.match_locale('fr-be', locale_tag_list), 'fr-BE')
-        self.assertEqual(locales.match_locale('fr-ca', locale_tag_list), 'fr-CA')
-        self.assertEqual(locales.match_locale('fr-ch', locale_tag_list), 'fr-CH')
-        self.assertEqual(locales.match_locale('zh-cn', locale_tag_list), 'zh-CN')
-        self.assertEqual(locales.match_locale('zh-tw', locale_tag_list), 'zh-TW')
-        self.assertEqual(locales.match_locale('zh-hk', locale_tag_list), 'zh-HK')
-
+        self.assertEqual(locales.match_locale(locale, self.locale_tag_list), expected_locale)
+
+    @parameterized.expand(
+        [
+            ('zh-hans', 'zh-CN'),
+            ('zh-hans-cn', 'zh-CN'),
+            ('zh-hant', 'zh-TW'),
+            ('zh-hant-tw', 'zh-TW'),
+        ]
+    )
+    def test_match_lang_script_code(self, locale: str, expected_locale: str):
         # Test language script code
+        self.assertEqual(locales.match_locale(locale, self.locale_tag_list), expected_locale)
 
-        self.assertEqual(locales.match_locale('zh-hans', locale_tag_list), 'zh-CN')
-        self.assertEqual(locales.match_locale('zh-hans-cn', locale_tag_list), 'zh-CN')
-        self.assertEqual(locales.match_locale('zh-hant', locale_tag_list), 'zh-TW')
-        self.assertEqual(locales.match_locale('zh-hant-tw', locale_tag_list), 'zh-TW')
-
-        # Test individual locale lists
-
-        self.assertEqual(locales.match_locale('es', [], fallback='fallback'), 'fallback')
-
+    def test_locale_de(self):
```

Comment:
The `test_local_xxx` for specific regions seems to be specific to that region, so I combined them by that name.